### PR TITLE
Forward CancellationToken

### PR DESCRIPTION
**Pull request by Youssef1313**
This fixes the cases that can't be detected by @carlossanlop's analyzer.
I've left comments in code for things I'm in doubt with.

### CODE REVIEW

**Comment by mavasani** [comment]
The token is not a parameter, but a property on the context type which is the parameter.

**Comment by mavasani** [comment]
I think we can enhance this analyzer in future to consider looking for properties of type CancellationToken on (struct) parameter types and have either a separate diagnostic ID or an editorconfig option to control this richer functionality. What do you think @carlossanlop?

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SyntaxNodeHelper.cs [hunk 1]
```diff
@@ -123,17 +123,17 @@ public IEnumerable<IMethodSymbol> GetCalleeMethodSymbols(SyntaxNode? node, Seman
             return GetDeclaredSymbol(node, semanticModel) ?? GetReferencedSymbol(node, semanticModel);
         }
 
-        public static ISymbol? GetDeclaredSymbol(SyntaxNode? node, SemanticModel semanticModel)
+        public static ISymbol? GetDeclaredSymbol(SyntaxNode? node, SemanticModel semanticModel) // TODO: Should I add an optional CancellationToken argument here to this public method?
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SyntaxNodeHelper.cs:126]: I don't think we want to spend any energy on analyzers in `Microsoft.NetFramework.Analyzers` project. Can you revert changes in this file and other files in NetFramework projects?

File: src/Utilities/Compiler/Extensions/IOperationExtensions.cs [hunk 2]
```diff
@@ -305,7 +305,7 @@ public static bool IsOperationNoneRoot(this IOperation operation)
             if (operation.Instance == null &&
                 operation.Property.ContainingType.IsAnonymousType)
             {
-                var declarationSyntax = operation.Property.ContainingType.DeclaringSyntaxReferences[0].GetSyntax();
+                var declarationSyntax = operation.Property.ContainingType.DeclaringSyntaxReferences[0].GetSyntax(); // TODO: Should the method take CancellationToken and pass it to GetSyntax() ?
```
Comment (mavasani) [inline src/Utilities/Compiler/Extensions/IOperationExtensions.cs:308]: Possibly, but I would leave that out of this PR and revert the change in this file.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs [hunk 3]
```diff
@@ -206,7 +206,7 @@ public override void Initialize(AnalysisContext analysisContext)
                         {
                             var semanticModel = argument.SemanticModel;
 
-                            var symbol = semanticModel.GetSymbolInfo(argument.Value.Syntax).Symbol;
+                            var symbol = semanticModel.GetSymbolInfo(argument.Value.Syntax, oaContext.CancellationToken).Symbol; // Does it matter here to use csaContext.CancellationToken or oaContext.CancellationToken?
```
Comment (Youssef1313) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs:209]: @mavasani, This is the only comment left now.

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs]: Please always use innermost context's cancellation token. Analyzer driver is free to invoke each analyzer callback with a different cancellation token, and in fact it does happen in the IDE execution. I have seen unhandled operation cancelled exceptions from analyzers not using the correct cancellation token.

**Comment by carlossanlop** [comment]
> I think we can enhance this analyzer in future to consider looking for properties of type CancellationToken on (struct) parameter types and have either a separate diagnostic ID or an editorconfig option to control this richer functionality. What do you think @carlossanlop?

I prefer the option of having a separate diagnostic ID.

**Comment by Youssef1313** [comment]
@mavasani @carlossanlop Anything left for this?
